File: 8-semester-pamo/exercise-3/Tipper/app/src/main/java/com/example/tipper/RecipeDetails.java
// Sandro Sobczyński (s20600)

package com.example.tipper;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class RecipeDetails extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.recipe_details);

        TextView recipeDetailsTextView = findViewById(R.id.recipeDetailsTextView);

        Intent intent = getIntent();
        String recipeText = intent.getStringExtra("recipeText");

        if (recipeText != null && !recipeText.isEmpty()) {
            recipeDetailsTextView.setText(recipeText);
        } else {
            recipeDetailsTextView.setText("Brak szczegółów przepisu");
        }

        Button returnButton = findViewById(R.id.returnButton);
        returnButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(RecipeDetails.this, Eat.class);
                startActivity(intent);
            }
        });
    }


}

File: 8-semester-pamo/exercise-4/Tipper/app/src/main/java/com/example/tipper/RecipeDetails.kt
// Sandro Sobczyński (s20600)
package com.example.tipper

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class RecipeDetails : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.recipe_details)
        val recipeDetailsTextView = findViewById<TextView>(R.id.recipeDetailsTextView)
        val intent = intent
        val recipeText = intent.getStringExtra("recipeText")
        if (recipeText != null && !recipeText.isEmpty()) {
            recipeDetailsTextView.text = recipeText
        } else {
            recipeDetailsTextView.text = "Brak szczegółów przepisu"
        }
        val returnButton = findViewById<Button>(R.id.returnButton)
        returnButton.setOnClickListener {
            val intent = Intent(this@RecipeDetails, Eat::class.java)
            startActivity(intent)
        }
    }
}